src/main/java/ExecutorServiceSolution.java

```diff
@@ -0,0 +1,30 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class ExecutorServiceSolution {
+    public int solution(List<Integer> list) {
+        List<Callable<Integer>> callables = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            callables.add(new MyCallable(list, i * 100_000, (i + 1) * 100_000));
+        }
+
+        ExecutorService executorService = Executors.newFixedThreadPool(10);
```

Comment:
Don’t use magic numbers, let’s create constants instead.
```suggestion
        for (int i = 0; i < 10; i++) {
            callables.add(new MyCallable(list, i * 100_000, (i + 1) * 100_000));
        }

        ExecutorService executorService = Executors.newFixedThreadPool(10);
```